paf-mvp-operator-express/tests/integration-tests/operator.integration.write.test.ts

```diff
@@ -0,0 +1,341 @@
+        },
+      };
+
+      const { server, startMock, endMock } = await getContext(exceptionValidator);
+
+      const operatorClient = new ClientBuilder().setClientHost('no-permission.com').build(publicKeyProvider);
+      if (isRedirect) {
+        response = await supertest(server)
+          .get(await getRedirectWriteUrl(operatorClient))
+          .set('referer', refererUrl)
+          .set('Origin', refererUrl);
+      } else {
+        const writeRequest = await getRestWriteRequest(operatorClient);
+        response = await supertest(server)
+          .post(writeRequest.url)
+          .type('text/plain')
+          .send(JSON.stringify(writeRequest.payload));
+      }
+
+      assertError(response, 500, NodeErrorType.UNKNOWN_ERROR);
+      if (isRedirect) {
+        expect(removeQueryString(getRedirectUrl(response))).toEqual(refererUrl);
+      }
+      expect(startMock).toHaveBeenCalled();
+      expect(endMock).toHaveBeenCalled();
+    });
+    it('should check payload', async () => {
+      const { server, startMock, endMock } = await getContext();
+
+      const response = isRedirect
+        ? await supertest(server)
+            .get('/paf/v1/redirect/post-ids-prefs')
+            .set('referer', refererUrl)
+            .set('Origin', refererUrl)
+        : await supertest(server).post('/paf/v1/ids-prefs');
+
+      assertError(response, 400, isRedirect ? NodeErrorType.INVALID_QUERY_STRING : NodeErrorType.INVALID_JSON_BODY);
+      if (isRedirect) {
+        expect(removeQueryString(getRedirectUrl(response))).toEqual(refererUrl);
+      }
+      expect(startMock).toHaveBeenCalled();
+      expect(endMock).toHaveBeenCalled();
+    });
+    it('should check permissions', async () => {
+      const { server, startMock, endMock } = await getContext();
+
+      const operatorClient = new ClientBuilder().setClientHost('paf.read-only.com').build(publicKeyProvider);
+
+      if (isRedirect) {
+        const writeUrl = await getRedirectWriteUrl(operatorClient);
+        response = await supertest(server).get(writeUrl).set('referer', refererUrl).set('Origin', refererUrl);
+      } else {
+        const writeRequest = await getRestWriteRequest(operatorClient);
+        response = await supertest(server)
+          .post(writeRequest.url)
+          .type('text/plain')
+          .send(JSON.stringify(writeRequest.payload));
+      }
+
+      assertError(response, 403, NodeErrorType.UNAUTHORIZED_OPERATION);
+      if (isRedirect) {
+        expect(removeQueryString(getRedirectUrl(response))).toEqual(refererUrl);
+      }
+      expect(startMock).toHaveBeenCalled();
+      expect(endMock).toHaveBeenCalled();
+    });
+
+    describe('should check message signature', () => {
+      it('for wrong signature', async () => {
+        const { server, startMock, endMock } = await getContext();
+        const operatorClient = new ClientBuilder().setClientPrivateKey(specificPrivateKey).build(publicKeyProvider);
+        if (!isRedirect) {
```

Comment:
NIT: don't you think it would be more readable to switch all the `if/else` to have `if (isRedirect)` first?

Reply:
Sorry, I don't understand how you handled this comment?

Reply:
done!